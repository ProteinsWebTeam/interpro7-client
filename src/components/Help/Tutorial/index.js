import React, { PureComponent } from 'react';
import T from 'prop-types';

import Link from 'components/generic/Link';

import loadable from 'higherOrder/loadable';
import loadData from 'higherOrder/loadData';
import { getUrlForMeta } from 'higherOrder/loadData/defaults';

import { schemaProcessDataForDB } from 'schema_org/processors';

import { foundationPartial } from 'styles/foundation';

import ipro from 'styles/interpro-new.css';
import fonts from 'EBI-Icon-fonts/fonts.css';
import local from './style.css';

const f = foundationPartial(local, fonts, ipro);

const SchemaOrgData = loadable({
  loader: () => import(/* webpackChunkName: "schemaOrg" */ 'schema_org'),
  loading: () => null,
});

/*:: type Props = {
  data: {
    loading: boolean,
    payload?: {
      databases: {},
    }
  },
}; */

export const Tutorial = class extends PureComponent /*:: <Props> */ {
  static displayName = 'Tutorial';

  static propTypes = {
    data: T.shape({
      loading: T.bool.isRequired,
      payload: T.shape({
        databases: T.object,
      }),
    }).isRequired,
  };

  render() {
    const { loading, payload } = this.props.data;
    if (loading || !payload) return 'Loadingâ€¦';
    return (
      <section>
        <h3>Tutorials</h3>
        <p>A number of online tutorials relating to InterPro are available.</p>
        <div className={f('flex-grid', 'tutorials')}>
          <div className={f('card-grid', 'tuto-method')}>
            <Link href="//www.ebi.ac.uk/training/online/course/interpro-case-study-3-protein-family-building-methodologies">
              <div className={f('card-image')}>
                <div className={f('card-label')}>Tutorial</div>
              </div>
              <div className={f('card-content')}>
                <div className={f('card-title')}>
                  InterPro: A case study of 3 protein family building
                  methodologies
                </div>
                <div className={f('card-info')}>
                  <div className={f('card-info-author')}>
                    <span
                      className={f('icon', 'icon-common')}
                      data-icon="&#x235;"
                    />{' '}
                    Lorna Richardson, Shoshana Brown, Paul Thomas, David Haft{' '}
                  </div>
                  <div className={f('card-info-level')}>
                    <span
                      className={f('icon', 'icon-common')}
                      data-icon="&#x14e;"
                    />{' '}
                    Beginner
                  </div>
                  <div className={f('card-info-duration')}>
                    <span
                      className={f('icon', 'icon-common')}
                      data-icon="&#x17;"
                    />{' '}
                    1 hour
                  </div>
                </div>
                <div className={f('card-description')}>
                  This course provides an insight into the different protein
                  family building methodologies employed by three InterPro
                  member databases, namely SFLD (Structure Function Linkage
                  Database), TIGRFAMs and PANTHER. It also describes an
                  automated approach to comparing protein signatures and
                  predicting hierarchical relationships between them.
                </div>
              </div>
              {
                // <Link  href="//www.ebi.ac.uk/training/online/course/interpro-case-study-3-protein-family-building-methodologies" className={f('button')}>Start the course</Link>
              }
            </Link>
          </div>

          <div className={f('card-grid', 'tuto-what')}>
            <Link href="//www.ebi.ac.uk/training/online/course/interpro-functional-and-structural-analysis-protei/what-interpro">
              <div className={f('card-image')}>
                <div className={f('card-label')}>Tutorial</div>
              </div>

              <div className={f('card-content')}>
                <div className={f('card-title')}>
                  InterPro: functional and structural analysis of protein
                  sequences
                </div>
                <div className={f('card-info')}>
                  <div className={f('card-info-author')}>
                    <em
                      className={f('icon', 'icon-common')}
                      data-icon="&#x235;"
                    />{' '}
                    Alex Mitchell
                  </div>
                  <div className={f('card-info-level')}>
                    <em
                      className={f('icon', 'icon-common')}
                      data-icon="&#x14e;"
                    />{' '}
                    Beginner
                  </div>
                  <div className={f('card-info-duration')}>
                    <em
                      className={f('icon', 'icon-common')}
                      data-icon="&#x17;"
                    />{' '}
                    1 hour
                  </div>
                </div>
                <div className={f('card-description')}>
                  This course will provide an in-depth guide to InterPro - how
                  to access it, how to search the database and interpret the
                  results.
                </div>
              </div>
              {
                // <Link  href="//www.ebi.ac.uk/training/online/course/interpro-functional-and-structural-analysis-protei/what-interpro" className={f('button')}>Start the course</Link>
              }
            </Link>
          </div>

          <div className={f('card-grid', 'tuto-intro')}>
            <Link href="//www.ebi.ac.uk/training/online/course/protein-classification-introduction-embl-ebi-resou">
              <div className={f('card-image')}>
                <div className={f('card-label')}>Tutorial</div>
              </div>
              <div className={f('card-content')}>
                <div className={f('card-title')}>
                  Protein classification: An introduction to EMBL-EBI resources
                </div>
                <div className={f('card-info')}>
                  <div className={f('card-info-author')}>
                    <em
                      className={f('icon', 'icon-common')}
                      data-icon="&#x235;"
                    />{' '}
                    Amaia Sangrador{' '}
                  </div>
                  <div className={f('card-info-level')}>
                    <em
                      className={f('icon', 'icon-common')}
                      data-icon="&#x14e;"
                    />{' '}
                    Beginner
                  </div>
                  <div className={f('card-info-duration')}>
                    <em
                      className={f('icon', 'icon-common')}
                      data-icon="&#x17;"
                    />{' '}
                    0.5 hour
                  </div>
                </div>
                <div className={f('card-description')}>
                  This course provides background information on protein
                  signature databases and their use in sequence analysis and
                  protein classification.
                </div>
              </div>
              {
                //<Link  href="//www.ebi.ac.uk/training/online/course/protein-classification-introduction-embl-ebi-resou" className={f('button')}>Start the course</Link>
              }
            </Link>
          </div>

          <div className={f('card-grid', 'tuto-tour')}>
            <Link href="//www.ebi.ac.uk/training/online/course/interpro-quick-tour">
              <div className={f('card-image')}>
                <div className={f('card-label')}>Tutorial</div>
              </div>
              <div className={f('card-content')}>
                <div className={f('card-title')}>A Quick Tour of InterPro</div>
                <div className={f('card-info')}>
                  <div className={f('card-info-author')}>
                    <em
                      className={f('icon', 'icon-common')}
                      data-icon="&#x235;"
                    />{' '}
                    Alex Mitchell{' '}
                  </div>
                  <div className={f('card-info-level')}>
                    <em
                      className={f('icon', 'icon-common')}
                      data-icon="&#x14e;"
                    />{' '}
                    Beginner
                  </div>
                  <div className={f('card-info-duration')}>
                    <em
                      className={f('icon', 'icon-common')}
                      data-icon="&#x17;"
                    />{' '}
                    0.5 hour
                  </div>
                </div>
                <div className={f('card-description')}>
                  This course provides a brief overview of InterPro, including
                  querying the database and visualising the results using the
                  web interface.
                </div>
              </div>
              {
                // <Link  href="//www.ebi.ac.uk/training/online/course/interpro-quick-tour" className={f('button')}>Start the course</Link>
              }
            </Link>
          </div>
        </div>
        <h3 className={f('margin-top-large')}>Training</h3>
        <p>
          InterPro hands-on practical sessions feature regularly as part of the
          EBI training courses and workshops.{' '}
          <Link href="//www.ebi.ac.uk/training" className={f('ext')}>
            Find out more
          </Link>{' '}
          about when and where these training courses will run, as well as how
          to apply to host them.
        </p>
      </section>
    );
  }
};

export default loadData(getUrlForMeta)(Tutorial);
