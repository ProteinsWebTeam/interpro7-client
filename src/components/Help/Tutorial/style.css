/* grid not yet working with IE11*/
@import '../../../styles/colors.css';

.flex-grid {
  margin: 0;
  list-style-type: none;
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: repeat(auto-fill, minmax(240px, 1fr));
}

.image-tuto-method {
  background: url('../../../images/cover/tuto/photo_tuto_casestudy.png')
    no-repeat 88% 46%;
  background-size: 100%;
}

.image-tuto-what {
  background: url('../../../images/cover/tuto/sequence.png') no-repeat 50% 40%;
  background-size: 100%;
}

.image-tuto-intro {
  background: url('../../../images/cover/tuto/classification.png') no-repeat 50%
    40%;
  background-size: 100%;
}
.image-tuto-tour {
  background: url('../../../images/cover/tuto/quicktour.jpg') no-repeat 42% 40%;
  background-size: 190%;
}
.image-tuto-g3d {
  background: url('../../../images/cover/tuto/genome3d.png') no-repeat 45% 43%;
  background-size: 100%;
}

.box-add {
  border-left: 4px solid var(--colors-submain);
  padding: 0 1rem;
}
.box-add ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  line-height: 2;
}
