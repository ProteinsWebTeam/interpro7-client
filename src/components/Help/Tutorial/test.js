// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Tutorial from '.';

const renderer = new ShallowRenderer();

describe('<Tutorial />', () => {
  test('should render', () => {
    renderer.render(<Tutorial />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
