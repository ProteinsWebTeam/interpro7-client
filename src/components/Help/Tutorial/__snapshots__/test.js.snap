// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tutorial /> should render 1`] = `
<section>
  <h3>
    Tutorials
  </h3>
  <p>
    A number of online tutorials relating to InterPro are available.
  </p>
  <div
    className="flex-column"
  >
    <div
      className="card-grid"
    >
      <Connect(Link)
        href="//www.ebi.ac.uk/training/online/course/interpro-case-study-3-protein-family-building-methodologies"
        target="_blank"
      >
        <div
          className="card-image tuto-method"
        >
          <div
            className="card-tag tag-tuto"
          >
            Tutorial
          </div>
        </div>
        <div
          className="card-content"
        >
          <div
            className="card-title"
          >
            <h4>
              InterPro: A case study of 3 protein family building methodologies
            </h4>
          </div>
          <div
            className="card-info"
          >
            <div
              className="card-info-author"
            >
              <span
                className="icon icon-common"
                data-icon=""
              />
               
              Lorna Richardson, Shoshana Brown, Paul Thomas, David Haft
               
            </div>
            <div
              className="card-info-level"
            >
              <span
                className="icon icon-common"
                data-icon=""
              />
               
              Beginner
            </div>
            <div
              className="card-info-duration"
            >
              <span
                className="icon icon-common"
                data-icon=""
              />
               
              1 hour
            </div>
          </div>
          <div
            className="card-description"
          >
            This course provides an insight into the different protein family building methodologies employed by three InterPro member databases, namely SFLD (Structure Function Linkage Database), TIGRFAMs and PANTHER. It also describes an automated approach to comparing protein signatures and predicting hierarchical relationships between them.
          </div>
        </div>
        <div
          className="card-more"
        >
          <div
            className="button hollow secondary button-more icon icon-common icon-right margin-bottom-none"
            data-icon=""
          >
            Start the course
          </div>
        </div>
      </Connect(Link)>
    </div>
    <div
      className="card-grid"
    >
      <Connect(Link)
        href="//www.ebi.ac.uk/training/online/course/interpro-functional-and-structural-analysis-protei/what-interpro"
        target="_blank"
      >
        <div
          className="card-image tuto-what"
        >
          <div
            className="card-tag tag-tuto"
          >
            Tutorial
          </div>
        </div>
        <div
          className="card-content"
        >
          <div
            className="card-title"
          >
            <h4>
              InterPro: functional and structural analysis of protein sequences
            </h4>
          </div>
          <div
            className="card-info"
          >
            <div
              className="card-info-author"
            >
              <em
                className="icon icon-common"
                data-icon=""
              />
               
              Alex Mitchell
            </div>
            <div
              className="card-info-level"
            >
              <em
                className="icon icon-common"
                data-icon=""
              />
               
              Beginner
            </div>
            <div
              className="card-info-duration"
            >
              <em
                className="icon icon-common"
                data-icon=""
              />
               
              1 hour
            </div>
          </div>
          <div
            className="card-description"
          >
            This course will provide an in-depth guide to InterPro - how to access it, how to search the database and interpret the results.
          </div>
        </div>
        <div
          className="card-more"
        >
          <div
            className="button hollow secondary button-more icon icon-common icon-right margin-bottom-none"
            data-icon=""
          >
            Start the course
          </div>
        </div>
      </Connect(Link)>
    </div>
    <div
      className="card-grid"
    >
      <Connect(Link)
        href="//www.ebi.ac.uk/training/online/course/protein-classification-introduction-embl-ebi-resou"
        target="_blank"
      >
        <div
          className="card-image tuto-intro"
        >
          <div
            className="card-tag tag-tuto"
          >
            Tutorial
          </div>
        </div>
        <div
          className="card-content"
        >
          <div
            className="card-title"
          >
            <h4>
              Protein classification: An introduction to EMBL-EBI resources
            </h4>
          </div>
          <div
            className="card-info"
          >
            <div
              className="card-info-author"
            >
              <em
                className="icon icon-common"
                data-icon=""
              />
               
              Amaia Sangrador
               
            </div>
            <div
              className="card-info-level"
            >
              <em
                className="icon icon-common"
                data-icon=""
              />
               
              Beginner
            </div>
            <div
              className="card-info-duration"
            >
              <em
                className="icon icon-common"
                data-icon=""
              />
               
              0.5 hour
            </div>
          </div>
          <div
            className="card-description"
          >
            This course provides background information on protein signature databases and their use in sequence analysis and protein classification.
          </div>
        </div>
        <div
          className="card-more"
        >
          <div
            className="button hollow secondary button-more icon icon-common icon-right margin-bottom-none"
            data-icon=""
          >
            Start the course
          </div>
        </div>
      </Connect(Link)>
    </div>
    <div
      className="card-grid"
    >
      <Connect(Link)
        href="//www.ebi.ac.uk/training/online/course/interpro-quick-tour"
        target="_blank"
      >
        <div
          className="card-image tuto-tour"
        >
          <div
            className="card-tag tag-tuto"
          >
            Tutorial
          </div>
        </div>
        <div
          className="card-content"
        >
          <div
            className="card-title"
          >
            <h4>
              A Quick Tour of InterPro
            </h4>
          </div>
          <div
            className="card-info"
          >
            <div
              className="card-info-author"
            >
              <em
                className="icon icon-common"
                data-icon=""
              />
               
              Alex Mitchell
               
            </div>
            <div
              className="card-info-level"
            >
              <em
                className="icon icon-common"
                data-icon=""
              />
               
              Beginner
            </div>
            <div
              className="card-info-duration"
            >
              <em
                className="icon icon-common"
                data-icon=""
              />
               
              0.5 hour
            </div>
          </div>
          <div
            className="card-description"
          >
            This course provides a brief overview of InterPro, including querying the database and visualising the results using the web interface.
          </div>
        </div>
        <div
          className="card-more"
        >
          <div
            className="button hollow secondary button-more icon icon-common icon-right margin-bottom-none"
            data-icon=""
          >
            Start the course
          </div>
        </div>
      </Connect(Link)>
    </div>
  </div>
  <h3
    className="margin-top-large"
  >
    Training
  </h3>
  <p>
    InterPro hands-on practical sessions feature regularly as part of the EBI training courses and workshops.
     
    <Connect(Link)
      className="ext"
      href="//www.ebi.ac.uk/training"
    >
      Find out more
    </Connect(Link)>
     
    about when and where these training courses will run, as well as how to apply to host them.
  </p>
</section>
`;
